openlibrary/components/MergeUI/utils.js

```diff
@@ -230,6 +230,40 @@ function save_many(items, comment, action, data) {
     });
 }
 
+/**
+ * Fetches name associated with the author key
+ * @param {Object[]} works
+ * @returns {Promise<Record<string,object>} A response to the request
+ */
+export async function get_author_names(works) {
+    const authorIds = _.uniq(works).flatMap(record =>
+        record.authors.map(authorEntry => authorEntry.author.key)
```

Comment:
Fallback here in case authors is undefined
```suggestion
        (record.authors || []).map(authorEntry => authorEntry.author.key)
```